v2/index_gen.go

```diff
@@ -19,7 +19,12 @@ import (
 
 // GenerateIndex generates index for a given car in v1 format.
 // The index can be stored using index.Save into a file or serialized using index.WriteTo.
-func GenerateIndex(v1r io.Reader) (index.Index, error) {
+func GenerateIndex(v1r io.Reader, opts ...ReadOption) (index.Index, error) {
```

Comment:
I briefly wondered if GenerateIndex should take a ReadOptions struct, given that its caller has usually already applied the opts list. But I think that would make the API clunkier, and options are generally trivially cheap to apply.